lib/src/v2/crowd_notifier_primitives.ts

```diff
@@ -26,6 +28,47 @@ export function setupHA(): IKeyPair {
   return crypto_box_keypair();
 }
 
+/**
+ * Generates the static part for v2.1 from a pass phrase. The idea is to
+ * have a static part that doesn't change, to secure the notifications without
+ * having to keep too many keys.
+ * @param pkh public key of the health authority
+ * @param pp pass phrase, if possible 256 bits entropy or more
+ */
+export function genOrgStatic(pkh: Uint8Array, pp: string): IOrganizerData {
+  const mskO = new mcl.Fr();
+  mskO.setHashOf(from_string(pp));
+  const mpkO = mcl.mul(baseG2(), mskO);
+
+  const [mpkha, mskha] = keyGen();
+  const mpk = mcl.add(mpkO, mpkha);
+
+  const ctxtha = crypto_box_seal(mskha.serialize(), pkh);
+
+  return {mpk, mskO, ctxtha};
+}
+
+/**
+ * Generates the code for v2.1 with an organization master key.
+ * @param org from the genOrgStatic
+ * @param info about the room
+ * @return the QRCodeContent to be printed in QR codes
+ */
+export function genOrgCode(org: IOrganizerData, info: Uint8Array):
+    ILocationData {
+  const nonce1 = randombytes_buf(NONCE_LENGTH);
+  const nonce2 = randombytes_buf(NONCE_LENGTH);
+
+  const pEnt = {nonce1, nonce2};
+  const mtr = {
+    mpk: org.mpk,
+    mskl: org.mskO,
```

Comment:
why not use `mskl` instead of `msk0`?

Reply:
(never mind, `mskO` (as in Organization), makes sense to me)

Reply:
Still copying stuff from your documents - to be sure I'll understand it later and know where it comes from... ;)